src/app/clusters/general-commissioning-server/general-commissioning-server.cpp

```diff
@@ -328,6 +375,24 @@ bool emberAfGeneralCommissioningClusterSetRegulatoryConfigCallback(app::CommandH
     return true;
 }
 
+bool emberAfGeneralCommissioningClusterSetTCAcknowledgementsCallback(
+    chip::app::CommandHandler * commandObj, const chip::app::ConcreteCommandPath & commandPath,
+    const chip::app::Clusters::GeneralCommissioning::Commands::SetTCAcknowledgements::DecodableType & commandData)
+{
+#if CHIP_CONFIG_TC_REQUIRED
+    MATTER_TRACE_SCOPE("SetTCAcknowledgements", "GeneralCommissioning");
+    Commands::SetTCAcknowledgementsResponse::Type response;
+    EnhancedSetupFlowProvider * const enhancedSetupFlowProvider = Server::GetInstance().GetEnhancedSetupFlowProvider();
+    uint16_t acknowledgements                                   = commandData.TCUserResponse;
+    uint16_t acknowledgementsVersion                            = commandData.TCVersion;
+    CheckSuccess(enhancedSetupFlowProvider->SetTermsAndConditionsAcceptance(acknowledgements, acknowledgementsVersion), Failure);
```

Comment:
This is missing the check that the terms and conditions have to be accepted sufficiently  before taking them in.

> 11.10.7.8:
> 
> This command SHALL copy the user responses and accepted version to the presented Enhanced Setup Flow Terms & Conditions from the values provided in the TCUserResponse and TCVersion fields to the TCAcknowledgements Attribute and the TCAcceptedVersion Attribute fields respectively.
> 
> Success or failure of this command SHALL be communicated by the SetTCAcknowledgementsResponse command.
> 
> This command SHALL result in success with an ErrorCode value of OK in the SetTCAcknowledgementsResponse if all required terms were accepted by the user. Specifically, all bits have a value of 1 in TCAcknowledgements whose ordinal is marked as required in the file located at EnhancedSetupFlowTCUrl.
> 
> If the TCVersion field is less than the TCMinRequiredVersion, then the ErrorCode of TCMinVersionNotMet SHALL be returned and TCAcknowledgementsRequired SHALL remain unchanged.
> 
> If the TCUserResponse value indicates that all required terms were not accepted by the user, then the ErrorCode of RequiredTCNotAccepted SHALL be returned and TCAcknowledgementsRequired SHALL remain unchanged.
> 
>